RFS/src/main/java/com/rfs/version_es_6_8/SnapshotReader_ES_6_8.java

```diff
@@ -0,0 +1,80 @@
+package com.rfs.version_es_6_8;
+
+import org.opensearch.migrations.Version;
+import org.opensearch.migrations.VersionMatchers;
+import org.opensearch.migrations.cluster.ClusterSnapshotReader;
+
+import com.rfs.common.SnapshotRepo;
+import com.rfs.common.SourceRepo;
+import com.rfs.models.GlobalMetadata;
+import com.rfs.models.IndexMetadata;
+import com.rfs.models.ShardMetadata;
+
+public class SnapshotReader_ES_6_8 implements ClusterSnapshotReader {
+
+    private Version version;
+    private SourceRepo sourceRepo;
+
+    @Override
+    public boolean compatibleWith(Version version) {
+        return VersionMatchers.isES_6_8.test(version);
+    }
+
+    @Override
+    public void initialize(SourceRepo sourceRepo) {
+        this.sourceRepo = sourceRepo;
+    }
+
+    @Override
+    public GlobalMetadata.Factory getGlobalMetadata() {
+        return new GlobalMetadataFactory_ES_6_8(getSnapshotRepo());
+    }
+
+    @Override
+    public IndexMetadata.Factory getIndexMetadata() {
+        return new IndexMetadataFactory_ES_6_8(getSnapshotRepo());
+    }
+
+    @Override
+    public ShardMetadata.Factory getShardMetadata() {
+        return new ShardMetadataFactory_ES_6_8(getSnapshotRepo());
+    }
+
+    @Override
+    public int getBufferSizeInBytes() {
+        return ElasticsearchConstants_ES_6_8.BUFFER_SIZE_IN_BYTES;
+    }
+
+    @Override
+    public boolean getSoftDeletesPossible() {
+        return ElasticsearchConstants_ES_6_8.SOFT_DELETES_POSSIBLE;
+    }
+
+    @Override
+    public String getSoftDeletesFieldData() {
+        return ElasticsearchConstants_ES_6_8.SOFT_DELETES_FIELD;
+    }
+
+    @Override
+    public Version getVersion() {
+        return version;
+    }
+
+    @Override
+    public void initialize(Version version) {
+        this.version = version;
+    }
```

Comment:
It's a bit surprising to have this initialization option when it doesn't appear to be particularly useful for fulfilling the role of this type.  Can you help me understand the reasoning here?

Reply:
It wasn't needed until I wanted to start printing output where I needed user-provided/detected version rather than the implied versions of the Reader/Writer

Secondarily in the near future we'd want to print a warning message indicating our confidence in version mismatch.  E.g. We are found and are using SnapshotReader_ES_7_10, but the cluster version is ES 7.11.0.  We might want to provide templates for message unsupported features or errors when attempting to use 'off-label' versions.

Reply:
Ah, so we'll typically chain the two `initialize()` methods to get both values during object setup.  Got it!